js/app.js

```diff
@@ -1,29 +1,181 @@
-"use strict";
+const apiKey = "5b81342d";
+const searchForm = document.querySelector("#search-form");
+const resultsContainer = document.querySelector("#results");
+const detailsContainer = document.querySelector("#details");
+const paginationContainer = document.querySelector("#pagination");
+const modalOverlay = document.querySelector("#modal-overlay");
 
-const textDisplay = document.getElementById("text-display");
-const textEditor = document.getElementById("text-editor");
+/* waiting for user actions with the form */
+searchForm.addEventListener("submit", async (event) => {
+  event.preventDefault();
 
-window.addEventListener("keydown", actionOnEvent);
+  paginationContainer.style.display =
+    "none"; /* prevent pagination with no search result */
 
-let isEditing = false;
+  const title = document.querySelector("#title").value;
+  const type = document.querySelector("#type").value;
+  let page = 1;
 
-function actionOnEvent(event) {
-  // console.log(event);
-  if (event.ctrlKey && event.key === "e") {
-    event.preventDefault(); // Disable default behavior for Ctrl+E
-    if (!isEditing) {
-      textEditor.value = textDisplay.innerText;
-      textDisplay.style.display = "none";
-      textEditor.style.display = "block";
-      isEditing = true;
-    }
-  } else if (event.ctrlKey && event.key === "s") {
-    event.preventDefault(); // Disable default behavior for Ctrl+S
-    if (isEditing) {
-      textDisplay.innerText = textEditor.value;
-      textEditor.style.display = "none";
-      textDisplay.style.display = "block";
-      isEditing = false;
+  const movies = await getMovies(title, type, page);
+  displayResults(movies);
+});
+
+/* waiting for response with the movie data of the specified page */
+async function getMovies(title, type, page) {
+  const url = `https://www.omdbapi.com/?s=${encodeURIComponent(
+    title
+  )}&type=${encodeURIComponent(type)}&page=${page}&apikey=${apiKey}`;
+
+  const response = await fetch(url);
+  const data = await response.json();
+  const totalResults = data.totalResults;
+
+  console.log(data);
+
+  if (data.Response === "True") {
+    displayPagination(totalResults, title, type, page);
+
+    return data.Search;
+  } else {
+    return [];
+  }
+}
+
+/* displaying number of pages and controls */
+function displayPagination(totalResults, title, type, page) {
+  let currentPage = page;
+  let totalPages = Math.ceil(totalResults / 10);
+
+  if (totalPages > 100) {
+    totalPages = 100;
+  }
```

Comment:
`Math.min(100, Math.ceil(totalResults / 10))`

Reply:
Thank you! Done!